File: SMI-SDK/.NET SDK/Library/gis/trace/MessageEntry.cs
namespace pt.sapo.gis.trace
{
    using System;
    using System.Collections.Generic;
    using pt.sapo.gis.trace.configuration;

    /**
     * @author GIS Team
     * Class for entry with an message
     */
    public class MessageEntry : Entry
    {
        public static readonly String MESSAGE_PROPERTY_NAME = "message";

        public MessageEntryProperty getMessage()
        {
            return (MessageEntryProperty)Properties[MESSAGE_PROPERTY_NAME];
        }

        public void setMessage(MessageEntryProperty message)
        {
            Properties.Add(MESSAGE_PROPERTY_NAME, message);
        }

        public MessageEntry() { }

        public MessageEntry(String description, severityType severity) : base(description, severity) { }

        public MessageEntry(String description, severityType severity, IDictionary<String, Object> properties) : base(description, severity, properties) { }

    }
}

File: SMI-SDK/Java SDK/src/pt/sapo/gis/trace/MessageEntry.java
package pt.sapo.gis.trace;

import java.util.Map;
import pt.sapo.gis.trace.configuration.SeverityType;

/**
 * @author GIS Team
 * Class for entry with an message
 */
public class MessageEntry extends Entry{
    public static final String MESSAGE_PROPERTY_NAME = "message";
    
    public MessageEntryProperty getMessage() {
        return (MessageEntryProperty) getProperty(MESSAGE_PROPERTY_NAME);
    }

    public void setMessage(MessageEntryProperty message) {
        setProperty(MESSAGE_PROPERTY_NAME, message);
    }
    
    public MessageEntry(){
        super();
    }
    
     public MessageEntry(String description, SeverityType severity){    
        super(description, severity);
    }
    
    public MessageEntry(String description, SeverityType severity, Map<String,Object> properties){    
        super(description, severity, properties);
    }
       
}
